tests/io/eager/test_write.py

```diff
@@ -273,3 +274,22 @@ def test_write_single_partition_different_partitioning(store_factory):
         store=store_factory, dataset_uuid=dataset.uuid, data=new_data
     )
     assert initial_keys + 2 == len(list(store_factory().keys()))
+
+
+def test_store_dataframes_as_dataset_does_not_allow_invalid_indices(store_factory):
+    partitions = [
+        {
+            "label": "part1",
+            "data": [("core", pd.DataFrame({"p": [1, 2]}))],
+            "indices": {"x": ExplicitSecondaryIndex("x", {1: ["part1"], 2: ["part2"]})},
+        }
+    ]
+
+    with pytest.raises(ValueError) as err:
```

Comment:
Rather use `with pytest.raises(ValueError, match="In table core, no column corresponding to index x") than asserting the exception youself

Reply:
Done!